core/src/test/java/org/jclouds/io/internal/BasePayloadSlicerTest.java

```diff
@@ -71,6 +71,33 @@ public void testIterableSliceExpectedMulti() throws IOException {
    }
 
    @Test
+   public void testIterableSliceWithRepeatingByteSource0() throws IOException {
```

Comment:
I reworked the existing test and new test into `testIterableSliceWithRepeatingByteSourceSmallerPartSize` and `testIterableSliceWithRepeatingByteSourceLargerPartSize`.